File: RedSheet.DbEntities/Models/vProjectNegotionalityRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectNegotionalityRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectNegotionalityRecord  
	{
 
	[Required]
	public bool IsMarketDifficult { get; set; } 
 
	[Required]
	public bool IsSpendLarge { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int NegotionalityCategoryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectModuleId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int ProjectNegotionalityId { get; set; } 

	public vProjectNegotionalityRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-negotionality-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectNegotionalityRecord {
    constructor(vProjectNegotionalityRecord?: vProjectNegotionalityRecord )  {
        let properties = [ "isMarketDifficult", "isSpendLarge", "negotionalityCategoryId", "projectModuleId", "projectNegotionalityId",];
        for (let property of properties)
            if (vProjectNegotionalityRecord && vProjectNegotionalityRecord[property])
                this[property] = vProjectNegotionalityRecord[property];
    }
 
	isMarketDifficult : boolean = false ;
 
	isSpendLarge : boolean = false ;
 
    @range(1,2147483647)
	negotionalityCategoryId : number =   undefined;
 
	projectModuleId : number =   0 ;
 
    @range(1,2147483647)
	projectNegotionalityId : number =   undefined;


}
